packages/transaction-controller/src/utils/gas-fees.ts

```diff
@@ -12,33 +12,61 @@ import { GAS_ESTIMATE_TYPES } from '@metamask/gas-fee-controller';
 import { createModuleLogger } from '@metamask/utils';
 import { addHexPrefix } from 'ethereumjs-util';
 
+import { SWAP_TRANSACTION_TYPES } from '../constants';
 import { projectLogger } from '../logger';
-import type { TransactionParams } from '../types';
-import { UserFeeLevel, type TransactionMeta } from '../types';
+import type {
+  AdvancedGasFees,
+  TransactionParams,
+  TransactionMeta,
+  TransactionType,
+} from '../types';
+import { UserFeeLevel } from '../types';
 
 export type UpdateGasFeesRequest = {
   eip1559: boolean;
   ethQuery: EthQuery;
+  getAdvancedGasFee: () => AdvancedGasFees;
```

Comment:
I won't comment every instance but should we change all the callbacks and variables to refer to `SavedGasFees` for consistency?

Reply:
Renamed all